Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,190 @@ extension UnifiedSymbolGraph {
+        // └───────────────────────────┘  └───────────────────────────┘
+        //
+        // The unified relationships are still consistent (there's one overload-group symbol and all
+        // the overloaded symbols point to it), but `overloadC` has different OverloadData depending
+        // on which selector you're using: In one graph it's at index 3, but in the other graph it's
+        // at index 2. In this case we need to recalculate the overload indices based on an ordering
+        // that takes all of the overloaded symbols into account.
+        //
+        // Finally, in situation 3 the relationship graph looks like this:
+        //
+        // ┌───────────────────────────────────────────┐
+        // │                 overloadA (1) ─┬─► groupA │
+        // │ groupB ◄─┬─ (1) overloadB (2) ─┘          │
+        // │          └─ (2) overloadC                 │
+        // └───────────────────────────────────────────┘
+        //
+        // In addition to overloadB having different indices per graph, the unified relationships
+        // are also inconsistent: there are two overload groups that refer to different subsets of
+        // symbols! In this case, we need to recalculate the overload indices like in situation 2
+        // above, but we also need to determine which overload group to keep as the "canonical" one.
+        // Since overload group symbols are by definition going to be generated by SymbolKit, when
+        // they're introduced into a unified symbol graph we can remove any that don't describe a
+        // "unified overload group" in this way, along with any relationships that point to them.
+        //
+        // In all three situations, we can generalize the solution with the following steps:
+        //
+        // 1. For each overload group, check the constituent symbols to see if they're part of
+        //    another overload group. Collect the symbols and groups as you go.
+        //    a. If you find other overload groups, recursively check for other overload groups
+        //       until no new groups are found.
+        // 2. Sort the collected symbols with the same comparator as
+        //    `SymbolGraph.createOverloadGroupSymbols()`.
+        // 3. Find the overload group symbol matching the first overload in the resulting order, and
+        //    ensure that it contains `overloadOf` relationships to all the overloaded symbols.
+        // 4. For each symbol, create a new OverloadData instance referencing the unified overload
+        //    group and the computed index in the sort order. Save it in the symbol's `unifiedMixins`.
+        // 5. If there were other overload group symbols, remove them from the symbol graph, along
+        //    with any relationships that referenced them.
+
+        /// A mapping of overload group symbols to their individual overloads.
+        var overloadGroups: [String: Set<String>] = [:]
```

Comment:
Both of these `String` types are referencing symbol unique identifiers. I feel like adding type aliases here would add more clutter than it's worth, especially since this is only a variable binding within a function and these aliases would only be written once or twice.